web/src/components/l10n/LocaleSelector.test.jsx

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) [2024] SUSE LLC
+ *
+ * All Rights Reserved.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of version 2 of the GNU General Public License as published
+ * by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, contact SUSE LLC.
+ *
+ * To contact SUSE LLC about this file by physical or electronic mail, you may
+ * find current contact information at www.suse.com.
+ */
+
+import React from "react";
+import { act, screen, waitFor, within } from "@testing-library/react";
+import { plainRender } from "~/test-utils";
+import { LocaleSelector } from "~/components/l10n";
+
+const locales = [
+  { id: "es_ES", name: "Spanish", territory: "Spain" },
+  { id: "en_US", name: "English", territory: "United States" }
+];
+
+const onChange = jest.fn();
+
+describe("LocaleSelector", () => {
+  it("renders a selector for given locales displaying their name, territory, and id", () => {
+    plainRender(
+      <LocaleSelector locales={locales} aria-label="Available locales" />
+    );
+
+    const selector = screen.getByRole("grid", { name: "Available locales" });
+
+    const options = within(selector).getAllByRole("row");
+    expect(options.length).toEqual(locales.length);
+
+    within(selector).getByRole("row", { name: "Spanish Spain es_ES" });
+    within(selector).getByRole("row", { name: "English United States en_US" });
+  });
+
+  it("renders an input for filtering locales", async () => {
+    const { user } = plainRender(
+      <LocaleSelector locales={locales} aria-label="Available locales" />
+    );
+
+    const filterInput = screen.getByRole("search");
```

Comment:
:warning: **NOTE:** This made me realize that Agama is doing wrong in terms of accessibility and usability by not using proper `<label>`s for no other reason than _visual aesthetic_. This example is working just because there is only one input with the "search" role, but it would fail trying to get it by its name/label too:

```diff
-   const filterInput = screen.getByRole("search");
+   const filterInput = screen.getByRole("search", { name: /Filter by language/} );
```

Of course, taking a look at the UI **for us** and thanks to the placeholder it's clear the input functionality

![Screenshot from 2024-01-24 19-41-52](https://github.com/openSUSE/agama/assets/1691872/dbcf0e5e-7f97-4b61-bd10-9f8b6563788a)

but it might not be as obvious to other users.

Something to work on. But in a different PR.

---

From W3c (https://www.w3.org/WAI/tutorials/forms/instructions/#placeholder-text)

> While placeholder text provides valuable guidance for many users, **placeholder text is not a replacement for labels**

From MDN (https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-placeholder)

> **Warning:** Using a placeholder instead of a visible label harms accessibility and usability for many users including **older users and users with cognitive, mobility, fine motor skill and vision impairments**. Labels are better: they are always visible and they provide for a larger hit region to focus on the control.
> ...
> **Note:** Placeholders should only be used to show an example of the type of data that should be entered into a form; they don't replace a proper label. 


Other links, 

https://www.w3.org/WAI/GL/low-vision-a11y-tf/wiki/Placeholder_Research

---

/cc @joseivanlopez, @imobachgs

Reply:
Good point! Definitely, "innovative" UI approaches are quite ... dangerous :)